jdr/WFLY-13539-galleon_layer_jdr_support.adoc

```diff
@@ -0,0 +1,89 @@
+-h,--help             Display this message and exit
+-H,--host <arg>       hostname that the management api is bound to. (default: localhost)
+-p,--port <arg>       port that the management api is bound to. (default: 9990)
+-s,--protocol <arg>   Protocol that is used to connect. Can be remote, http or https (default: http)
+----
+
+The availability of a management interface is optional. If there is not a management interface available, the `jdr.sh` script is still able to extract some information invoking internally the embedded server.
+
+The goal of this feature is to supply a https://docs.wildfly.org/galleon/#_layers[Galleon layer] to give support to the JDR subsystem.
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13539[WFLY-13539 Add a Galleon layer for JBoss Diagnostic Reporting]
+* https://issues.redhat.com/browse/EAP7-1444[EAP7-1444 Galleon Layers in EAP XP 2]
+
+=== Related Issues
+* https://issues.redhat.com/browse/WFGP-177[WFGP-177 Create a Galleon package for the JBoss Diagnostic Reporting bin scripts]
+
+=== Dev Contacts
+
+* mailto:brian.stansberry@redhat.com[Brian Stansberry]
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+* mailto:jkasik@redhat.com[Jan Kasik]
+
+=== Testing By
+
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+WildFly, WildFly Galleon Plugins
+
+=== Other Interested Projects
+
+== Requirements
+
+* The galleon layer must provide the provisioning of the JDR subsystem.
+* The galleon layer will depend optionally on `secure-management` layer. This layer will allow the users to configure a secure management interface to connect to the server via `jdr.sh`. If a user wants to connect via CLI, they have to add `core-tools` layer to their provision configuration.
+* The bash scripts to generate the diagnosis information must be provisioned and be available at `$WFLY_HOME/bin/`.
+
+== Test Plan
+
+The test coverage of the Galleon layer added by this proposal is divided in two main groups:
+
+. Test the Galleon layer provision. https://github.com/wildfly/wildfly/blob/master/testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java[LayersTestCase] will be expanded to test the provision of the added layer when it is used isolated and with other layers. For each sort of provision, this test does the following:
+.. Verifies the provisioned modules are the expected ones.
+.. Verifies the provisioned server starts successfully.
+. Execution of existing tests on a minimum set of layers. Reuse the existing tests available on the WildFly test suite, which are directly testing features added by this layer, and execute them on a server installation provisioned with this layer.
+
+== Community Documentation
+
+Community documentation plan is adding the layer to https://docs.wildfly.org/19/Admin_Guide.html#wildfly-galleon-layers[WildFly Galleon layers] in the section it belongs to.
```

Comment:
Are not release notes more oriented to the product instead of the community version? My understanding is the WildFly release notes are the Jira link. If not, any suggestion what a release note should contain?

Also, take into account this one has not a valid meaning on WildFly, since it is more a feature for the product.

Reply:
> Are not release notes more oriented to the product instead of the community version

The Wildfly (community) template requires for Release Notes content, so I guess this is expected here, too. See https://github.com/wildfly/wildfly-proposals/blob/master/design-doc-template.adoc#release-note-content

>  If not, any suggestion what a release note should contain?

Just a short sentence (or two, meaning something quick) that gives the user an immediate very-high-level idea of what is the feature being proposed, e.g.: https://github.com/wildfly/wildfly-proposals/blob/master/microprofile/WFLY-12313_MicroProfile_OpenAPI_1.1.adoc#release-note-content

> Also, take into account this one has not a valid meaning on WildFly, since it is more a feature for the product.

Actually, apart from the above reasoning of mine about the template defining such a section, I do see that this is one more reason for adding it. The release notes from the product must in fact be taken from somewhere and I think this is the right place for a starting point. WDYT?

Reply:
Right, I missed the point that the release notes are even in the template. I added one.

Reply:
Thanks